src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/installers/AnyOfInstaller.java

```diff
@@ -0,0 +1,307 @@
+                } else {
+                    msg = Messages.AnyOfInstaller_loops_installers_1attempt(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                }
+            } else {
+                if (maxAttemptsPerInstaller > 1) {
+                    msg = Messages.AnyOfInstaller_loops_1installer_attempts(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                } else {
+                    msg = Messages.AnyOfInstaller_loops_1installer_1attempt(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                }
+            }
+        } else {
+            if (numberOfConfiguredOfInstallers > 1) {
+                if (maxAttemptsPerInstaller > 1) {
+                    msg = Messages.AnyOfInstaller_1loop_installers_attempts(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                } else {
+                    msg = Messages.AnyOfInstaller_1loop_installers_1attempt(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                }
+            } else {
+                if (maxAttemptsPerInstaller > 1) {
+                    msg = Messages.AnyOfInstaller_1loop_1installer_attempts(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                } else {
+                    msg = Messages.AnyOfInstaller_1loop_1installer_1attempt(wholeListAttempt, maxWholeListAttempts,
+                            indexOfConfiguredInstaller, numberOfConfiguredOfInstallers, nameOfConfiguredInstaller,
+                            installerAttempt, maxAttemptsPerInstaller, whatToReport);
+                }
+            }
+        }
+        // now log it
+        final PrintStream output = log.getLogger();
+        output.println(msg);
+    }
+
+    /**
+     * Descriptor for the {@link AnyOfInstaller}.
+     */
+    @Extension

```

Comment:
Good practice: Add `@Symbol(`anyOf`)` to new extension points so that they can be easily picked by Jenkins Configuration as Code plugin

Reply:
Ah, I had wondered what that code was for :grin:
Done.